### ISSUE TITLE
Typescript model dispatch problems

### ISSUE THREAD

**Issue by arvindsouza** [rematch/rematch#568]
I'm trying to use dispatch models according to the example in the docs:
https://github.com/rematch/rematch/blob/master/docs/recipes/typescript.md

however, when trying to pass the mapDispatch to the connect function, I'm getting the following errors:
**Types of parameters 'dispatch' and 'dispatch' are incompatible.**

**Type 'Dispatch<Action<any>>' is not assignable to type 'RematchDispatch<{ resetData: ModelConfig<{ requestStatus: any; error: string | null; }, { requestStatus: any; error: string | null; }>; }>'.**

 **Type 'Dispatch<Action<any>>' is not assignable to type 'ExtractRematchDispatchersFromModels<{ resetData: ModelConfig<{ requestStatus: any; error: string | null; }, { requestStatus: any; error: string | null; }>; }> & ((action: Action<void, void>) => Dispatch<...>) & ((action: Action<...>) => Dispatch<...>) & ((...args: any[]) => Action<...>) & Dispatch<...>'.**

Here's how I'm declaring mapDispatch and calling connect():
```
   return connect(null, mapDispatch )(Wrapper); //errors on mapDispatch
.
.
const mapDispatch = (dispatch: Dispatch) => ({
    sendData: (data: Action<string>) => dispatch.resetData.sendNewData(data),
});
```

**Comment by d3dc** [comment]
In your `mapDispatch` function, you can just return the dispatcher `dispatch.resetData.sendNewData`

I assume `Dispatch` is `typeof store.dispatch`. I'm not sure why you're getting an assignment error, it maybe be trying to match your new function against the shape of Dispatch? i.e. `sendNewData` isn't `Dispatch<Action<string>>`. I'm curious why you wrap your type in Action?

**Comment by arvindsouza** [comment]
returning it doesn't fix the issue. Is there any way to force Dispatch to match? Sorry if that's a dumb question, I'm kind of new to react/redux.

As for wrapping the type in Action, It was throwing an error: 
**[ts] Argument of type 'string' is not assignable to parameter of type 'Action<any, void>'.**
when I passed any other value. the function actually accepts a string value in the mode. This problem is present in the example code as well.

**Comment by d3dc** [comment]
This might be related to #552. I'm starting to look at ways to actually get the typings testable and there's a few intersection types that don't quite grok for me - dispatch is one of them for sure.

**Comment by Arelav** [comment]
I'm not sure if is the same but try `"strictNullChecks": true` in tsconfig.

**Comment by skurfuerst** [comment]
@d3dc @arvindsouza I am also trying to fix this issue. While I did not find a fix yet, #574 contains a test suite based on dtslint which reproduces the error.

**Comment by arvindsouza** [comment]
Thanks for the help, I just set it to 'dispatch: any' for the time being. Will probably be removing rematch altogether eventually.
